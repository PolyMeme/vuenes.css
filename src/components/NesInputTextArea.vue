<template>
  <div class="field">
    <label
      v-if="labelText"
      :for="id"
    >
      {{ labelText }}
    </label>
    <textarea
      :id="id"
      v-model="inputText"
      class="nes-textarea"
      :cols="cols"
      :rows="rows"
      @input.prevent="emitInput"
    />
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: String,
      required: true
    },
    labelText: {
      type: String,
      default: () => ''
    },
    initValue: {
      type: String,
      default: () => ''
    },
    cols: {
      type: Number,
      default: 30
    },
    rows: {
      type: Number,
      default: 10
    }
  },
  data () {
    return {
      inputText: this.initValue
    }
  },
  methods: {
    emitInput () {
      this.$emit('input', this.inputText)
    }
  }
}
</script>

<style lang="scss">
@import "~typeface-press-start-2p/index.css";
@import "~nes.css/scss/base/index";
@import "~nes.css/scss/form/inputs";
</style>
