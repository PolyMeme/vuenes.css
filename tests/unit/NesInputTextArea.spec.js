import { mount } from '@vue/test-utils'
import NesInputTextArea from '../../src/components/NesInputTextArea.vue'

describe('NesInputTextArea.vue', () => {
  const createTextArea = (propsData) => {
    return mount(NesInputTextArea, {
      propsData
    })
  }

  it('should render the text area properly', () => {
    const wrapper = createTextArea({
      id: 'input-test'
    })

    expect(wrapper.find('div.field>textarea#input-test.nes-textarea').exists()).toBeTruthy()
  })

  it('should emit input event', () => {
    const wrapper = createTextArea({
      id: 'input-test'
    })

    const textArea = wrapper.find('div.field>textarea#input-test.nes-textarea')
    textArea.element.value = 'This is a new Value'
    textArea.trigger('input')

    expect(wrapper.emitted('input')).toEqual([['This is a new Value']])
  })

  it('should match snapshot', () => {
    const wrapper = createTextArea({
      id: 'input-test',
      labelText: 'This is a label!',
      initValue: 'Init Value!',
      cols: 50,
      rows: 50
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
