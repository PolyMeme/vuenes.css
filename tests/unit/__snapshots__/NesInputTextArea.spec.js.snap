// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesInputTextArea.vue should match snapshot 1`] = `
<div class="field"><label for="input-test">
    This is a label!
  </label> <textarea id="input-test" cols="50" rows="50" class="nes-textarea"></textarea></div>
`;
